cpp/scripts/include_checker.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+import sys
+import re
+import os
+import subprocess
+import argparse
+
+
+IncludeRegex = re.compile(r"\s*#include\s*(\S+)")
+
+
+def parseArgs():
+    argparser = argparse.ArgumentParser(
+        "Checks for a consistent '#include' syntax")
+    argparser.add_argument("-regex", type=str,
+                           default=r"[.](cu|cuh|h|hpp|cpp)$",
+                           help="Regex string to filter in sources")
+    argparser.add_argument("dirs", type=str, nargs="*",
+                           help="List of dirs where to find sources")
+    args = argparser.parse_args()
+    args.regexCompiled = re.compile(args.regex)
+    return args
+
+
+def listAllSourceFiles(fileRegex, srcdirs):
+    allFiles = []
+    for srcdir in srcdirs:
+        for root, dirs, files in os.walk(srcdir):
+            for f in files:
+                if re.search(fileRegex, f):
+                    src = os.path.join(root, f)
+                    allFiles.append(src)
+    return allFiles
+
+
+def checkIncludesIn(src):
+    errs = []
+    fp = open(src, "r")
+    dir = os.path.dirname(src)
+    lNum = 0
+    for line in fp.readlines():
```

Comment:
not a big deal, but I think it'd be more idiomatic to do:

`for line in open(src):`

then you can omit the separate open and fp.close()

Reply:
better actually do do: ```
 for line_number, line in enumerate(open(src)):
```
then you don't have to do the lnum bookkeeping.

Reply:
thanks. this is much better! Done.